src/common/KokkosKernels_SparseUtils.hpp

```diff
@@ -315,40 +315,25 @@ template <typename in_row_view_t,
           typename out_nnz_view_t,
           typename out_scalar_view_t,
           typename tempwork_row_view_t,
-          typename MyExecSpace>
+          typename MyExecSpace,
+          bool transpose_values>
```

Comment:
@srajama1 I think it makes sense for that to be compile-time since it saves a branch in the innermost loop, and there's already different user-facing functions for graph vs. matrix.

I'm trying to figure out whether there's really a problem with the old transpose, or whether something in D2 is just making it look like there's a problem. With the old transpose impl, the test_bipartite with columns ends up with a tranpose rowmap that's all zero (obviously wrong). When I rewrote transpose, I got a plausible rowmap there and everything passed (including sparse unit tests, which calls transpose many times).

However, I wrote a unit test for transpose itself to try to prove there's a bug, but they all passed with the old implementation. So I'm still trying to figure out what's really going on with valgrind.